src/common/transformations/src/transformations/common_optimizations/sdpa_fusion.cpp

```diff
@@ -25,32 +26,75 @@ SDPAFusion::SDPAFusion() {
+    auto q = q_reshaped | q_base;
+
+    auto k_base = makePattern(ov::Rank(4));
+    auto k_shape = ov::pass::pattern::any_input();
+    auto k_reshaped = ov::pass::pattern::wrap_type<ov::op::v1::Reshape>({k_base, k_shape});
+    auto k = k_reshaped | k_base;
+
+    auto v_base = makePattern(ov::Rank(4));
+    auto v_proj_shape_m = ov::pass::pattern::any_input();
+    auto v_reshaped = ov::pass::pattern::wrap_type<ov::op::v1::Reshape>({v_base, v_proj_shape_m});
+    auto v = v_reshaped | v_base;
+
+    // Optional k scale
+    auto attn_scale = ov::pass::pattern::any_input();
+    // auto k_opt_scaled = optional<ov::op::v1::Multiply>({k, attn_scale});
+    // K transpose + optional scale
+    auto k_trans_dims = ov::pass::pattern::any_input();
+    auto k_opt_transposed = optional<ov::op::v1::Transpose>({k, k_trans_dims});
+    // auto k_transposed_opt_scaled = optional<ov::op::v1::Multiply>({_optk_transposed, attn_scale});
+    auto k_opt_transposed_scaled = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({k_opt_transposed, attn_scale});
+    auto k_opt_transposed_opt_scaled = k_opt_transposed_scaled | k_opt_transposed;
+    
+    // No transpose check here, there are scenarios where k is not transposed and that uses equation (A*B)^T = B^T * A^T
+    auto qk = makePattern<ov::op::v0::MatMul>({q, k_opt_transposed_opt_scaled});
+
+    // Optional unsqueeze that is converted to Reshape
+    auto unsqueeze_axis = ov::pass::pattern::wrap_type<ov::op::v0::Constant>();
+    auto qk_unsqueeze = ov::pass::pattern::wrap_type<ov::op::v1::Reshape>({qk, unsqueeze_axis});
+    auto qk_opt_unsqueeze = qk_unsqueeze | qk;
+
+    auto qk_scaled = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({qk_opt_unsqueeze, attn_scale});
+    auto qk_opt_scaled = qk_scaled | qk_opt_unsqueeze;
+
+    // optional mask add, there are patterns where before or/and after mask add buffer is reshaped
     auto mask = makePattern();
-
-    auto k_transpose_order = pattern::wrap_type<ov::op::v0::Constant>([](const Output<Node>& node) {
-        auto axis_order = ov::as_type_ptr<ov::op::v0::Constant>(node.get_node_shared_ptr())->cast_vector<int64_t>();
-        return axis_order == std::vector<int64_t>{0, 1, 3, 2};
-    });
-
-    auto k_t = pattern::wrap_type<ov::op::v1::Transpose>({k, k_transpose_order});
-    auto qk_nn = makePattern<ov::op::v0::MatMul>({q, k_t}, {{"transpose_a", false}, {"transpose_b", false}});
-    auto qk_nt = makePattern<ov::op::v0::MatMul>({q, k}, {{"transpose_a", false}, {"transpose_b", true}});
-    auto qk = qk_nt | qk_nn;
-    auto optional_add_mask = optional<ov::op::v1::Add>({qk, mask});
-    auto softmax = makePattern<ov::op::v8::Softmax>({optional_add_mask}, {{"axis", "-1"}});
-    auto qkv = makePattern<ov::op::v0::MatMul>({softmax, v}, {{"transpose_a", false}, {"transpose_b", false}});
+    // Optional reshape befor adding mask
+    auto qk_opt_scaled_pre_mask_shape = ov::pass::pattern::any_input();
```

Comment:
We have `using ov::pass::pattern` in the beginning of the function, so you can omit explicit namespace specification